File: YoutubeClone/DTO/LoginResponse.cs
namespace YoutubeClone.DTO
{
    public class LoginResponse
    {
        public UserSummary User { get; }
        public string Token { get; } = string.Empty;

        public LoginResponse(UserSummary userSummary, string token)
        {
            User = userSummary;
            Token = token;
        }
    }
}

File: YoutubeClone/ClientApp/src/api/models/login-response.ts
import { mapJsonToUserSummary, UserSummary } from "./user-summary";

export interface LoginResponse {
    user: UserSummary;
    token: string;
}

export function mapJsonToLoginResponse(json: any): LoginResponse {
    return {
        user: mapJsonToUserSummary(json.user),
        token: json.token
    }
}